app/src/main/java/com/onboarding/pronosticAR/adapter/WeatherAdapter.kt

```diff
@@ -0,0 +1,53 @@
+package com.onboarding.pronosticAR.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.onboarding.pronosticAR.R
+import com.onboarding.pronosticAR.databinding.ItemWeatherBinding
+import com.onboarding.pronosticAR.domain.entity.WeatherList
+import com.onboarding.pronosticAR.utils.ConstantUtils
+import com.onboarding.pronosticAR.utils.DateUtils
+import com.onboarding.pronosticAR.utils.WindDirectionConverter
+
+class WeatherAdapter(private val list: List<WeatherList>) : RecyclerView.Adapter<WeatherAdapter.ViewHolder>() {
+
+    override fun getItemCount() = list.size
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_weather, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(list[position])
+    }
+
+    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val binding = ItemWeatherBinding.bind(itemView)
```

Comment:
you are already using this val at with in line 30

Reply:
removed and used this instead